server/src/service/job-service.ts

```diff
@@ -83,6 +90,13 @@ export class JobService {
     }))
   }
 
+  getUploadService(uploadUrl: string): IUploadService {
+    if (s3UploadService.matchUrl(uploadUrl)) {
+      return s3UploadService
+    }
+    throw new BadRequestException(`Unsupported uploadUrl ${uploadUrl}`)
+  }
+
```

Comment:
I assume that S3 may not be the only place we upload images to in the future.
Request setup may differ between different cloud storages, so I use the URL provided by the client to determine which service the job should use.
Is this something we're expecting?